src/main/java/com/worksap/nlp/sudachi/Settings.java

```diff
@@ -73,41 +79,134 @@
+     *             if IO fails
+     * @see #parse(String, SettingsAnchor)
+     */
+    public Settings read(Path file) throws IOException {
+        logger.fine(() -> String.format("reading settings from %s", file));
+        String data = StringUtil.readFully(file);
+        Settings settings = parse(data, this.base);
+        return merge(settings);
+    }
+
+    /**
+     * Reads the content of the specified classpath resource and merges it into the
+     * current Settings object. Anchor is not modified.
+     *
+     * @param resource
+     *            result of {@link Class#getResource(String)}
+     * @return modified Settings object
+     * @throws IOException
+     *             if IO fails
+     * @see #parse(String, SettingsAnchor)
+     */
+    public Settings read(URL resource) throws IOException {
+        logger.fine(() -> String.format("reading settings from %s", resource));
+        String data = StringUtil.readFully(resource);
+        Settings settings = parse(data, this.base);
+        return merge(settings);
+    }
+
+    /**
+     * Read a settings from a JSON string.
+     * <p>
      *
      * @param path
-     *            the base path if "path" is undefined in {@code json}
+     *            will add additional {@link SettingsAnchor} to this path if not
+     *            {@code null}
      * @param json
      *            JSON string
-     * @return a structure of settings
+     * @return Settings object
      * @throws IllegalArgumentException
      *             if the parsing is failed
+     * @deprecated use {@link #parse(String, SettingsAnchor)}, this method will be
+     *             removed in 1.0.0
      */
+    @Deprecated
     public static Settings parseSettings(String path, String json) {
+        SettingsAnchor anchor = path == null ? SettingsAnchor.none() : SettingsAnchor.filesystem(Paths.get(path));
+        return parse(json, anchor);
+    }
+
+    /**
+     * Parse a JSON string into a Settings object. String must contain a JSON
+     * object. If a JSON contains a {@code path} key, its value will be prepended as
+     * an additional filesystem SettingsAnchor.
+     *
+     * @param json
+     *            JSON object as a String
+     * @param resolver
+     *            paths will be resolved against this {@link SettingsAnchor}
+     * @return SettingsObject
+     *
+     * @see SettingsAnchor#none()
+     * @see SettingsAnchor#filesystem(Path)
+     * @see SettingsAnchor#classpath()
+     */
+    public static Settings parse(String json, SettingsAnchor resolver) {
         try (JsonReader reader = Json.createReader(new StringReader(json))) {
             JsonStructure rootStr = reader.read();
             if (rootStr instanceof JsonObject) {
                 JsonObject root = (JsonObject) rootStr;
-                String basePath = root.getString("path", path);
-                return new Settings(root, basePath);
+                String basePath = root.getString("path", null);
```

Comment:
I thought that the in-config value should have higher priority than out-of-band configuration, so kept the current behavior, but I can change it. With the current structure of anchors, special treatment of the paths probably can be completely removed, but I am not sure because I am still not fully aware of all its usages.

Reply:
OK. Let's discuss this separately.